### PR TITLE
Listkeysv2

### PR DESCRIPTION
**Pull request by jreadey**
Use v2 interface for list keys (compatible with S3 express buckets)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -35,12 +35,11 @@ version = "0.8.5"
 
 dependencies = [
     "aiohttp == 3.9.4",
-    "aiobotocore == 2.5.0",
+    "aiobotocore == 2.13.0",
```
Comment (mattjala) [inline pyproject.toml:38]: `requirements.txt` should be updated as well

File: hsds/util/s3Client.py [hunk 2]
```diff
@@ -144,9 +147,17 @@
...
         kwargs["aws_secret_access_key"] = self._aws_secret_access_key
         kwargs["aws_access_key_id"] = self._aws_access_key_id
-        kwargs["aws_session_token"] = self._aws_session_token
-        kwargs["endpoint_url"] = self._s3_gateway
-        kwargs["use_ssl"] = self._use_ssl
+        if self._aws_session_token:
+            kwargs["aws_session_token"] = self._aws_session_token
+        if self._s3_gateway:
+            host = urlparse(self._s3_gateway).hostname
+            if not host.endswith("amazonaws.com"):
```
[the remark this answers is not in the document]
Comment (mattjala) [reply hsds/util/s3Client.py:154]: Replace with `.amazonaws.com`

File: pyproject.toml (hunk 1 above)
Comment (jreadey) [reply pyproject.toml:38]: done

**Review by mattjala — APPROVED** [review]
LGTM

**Pull request merged**